lib/matplotlib/axes/_base.py

```diff
@@ -220,9 +220,11 @@ def _xy_from_xy(self, x, y):
         x = _check_1d(x)
         y = _check_1d(y)
         if x.shape[0] != y.shape[0]:
-            raise ValueError("x and y must have same first dimension")
+            raise ValueError("x (shape: {}) and y (shape: {}) must have same "
+                             "first dimension".format(x.shape, y.shape))
```

Comment:
Thanks for this.  The double parens are a bit jarring though, and this gives more info than one needs.  Maybe:

```
"x and y must have same first dimension.  Got {} and {}.".format(x.shape[0], y.shape[0])
```